src/basilisp/lang/seq.py

```diff
@@ -174,7 +122,7 @@ def meta(self) -> Optional[IPersistentMap]:
         return self._meta
 
     def with_meta(self, meta: Optional[IPersistentMap]) -> "LazySeq[T]":
-        return LazySeq(self._gen, obj=self._obj, seq=self._seq, meta=meta)
+        return LazySeq(None, seq=self.seq(), meta=meta)
```

Comment:
It was never correct or safe to pass around internal `LazySeq` state like this. We needed to realize the seq to pass into the copied `LazySeq` to maintain thread safety.